### DISCUSSION TITLE
Strongswan -  simultaneously 2 type of auth

### DISCUSSION THREAD

**Discussion by MrStrangerMSK** [strongswan/strongswan#671] [Q&A]
Hi, all!
i have AWS setup for StrongSwan. 
I have successfully setup my win 10 connection to site, and now i try to get my Mikrotik router working
For Windows 10 setup i use EAP - MSChapv2 - so i use root certificate for server and than user auth.
Everything working.
But when i try to setup Mikrotik 4011 same way i can see certificate error while connection.
After that i try to setup simple PSK auth for mikrotik, but config doesnt selected.
My config is:
`include /var/lib/strongswan/ipsec.conf.inc

config setup
        uniqueids=never
        charondebug="ike 2, knl 2, cfg 2, net 2, esp 2, dmn 2,  mgr 2"
conn %default
        keyexchange=ikev2
	esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!
        fragmentation=yes
        rekey=no
        compress=yes
        dpdaction=clear
        leftsourceip=xx.xxx.xx.xx
        rightdns=8.8.8.8,8.8.4.4

conn WinUserCert
         left=%any
	 leftid=xx.xxx.xxx.xx
         leftsubnet=0.0.0.0/0
         leftcert=debian.pem
	 leftsendcert=always
         leftauth=pubkey
         right=%any
         rightsourceip=10.10.11.0/24
         rightauth=eap-mschapv2
         eap_identity=%identity
         auto=add
         dpdaction=clear
         dpddelay=300s
         ike=aes256-sha2_256-prfsha256-modp1024

conn Mikrotik-PSK
	type=tunnel
         leftauth=psk
         right=%any
         rightsourceip=10.10.10.0/24
         rightauth=psk
         auto=add
         dpdaction=clear
         dpddelay=300s
         ike=aes256-sha256-modp2048!
         esp=aes256-sha256-modp2048!
`
What should i do to make it work?

**Comment by tobiasbrunner** [comment]
Read the log (ideally with the log level or _cfg_ increased to 2).

Basically, both connections are equally valid (same settings for `left`, `right` and the IKE version, and the proposal probably matches too), so the first config will be used initially. If the peer does not send an identity (IDr) in the IKE_AUTH request and since you haven't configured `rightid` to match its `IDi`, there won't be a config switch later either.

Maybe you can change that, that is, either make the peer send an IDr or configure `rightid` to whatever identity the MikroTik box sends as its own, or configure the remote IP address in `right`. It might also work if you change the order of the configs and don't [fix Windows clients](https://wiki.strongswan.org/projects/strongswan/wiki/WindowsClients#AES-256-CBC-and-MODP2048) to use stronger DH groups (not recommended, though) as the IKE proposals sent by the Windows clients won't match the MikroTik config so the first config would get skipped for them.
